OpenNefia.Content/UI/Hud/HudLayer.cs

```diff
@@ -1,56 +1,371 @@
+            MessageBoxBacking.SetPosition(MinimapWidth, Height - MinimapHeight);
+            BacklogBacking.SetPosition(MinimapWidth + 10, Height - 470);
+            BacklogContainer.SetPosition(MinimapWidth + 15, BacklogBacking.Y + 12);
+            HudBar.SetPosition(MinimapWidth, Height - 18);
+            MessageBoxContainer.SetPosition(BacklogContainer.X, Height - MinimapHeight + 4);
+            DateText.SetPosition(120, 17);
+            ClockHand.SetPosition(62, 48);
+            HealthText.SetPosition(280, Height - MinimapHeight - 19);
+            ManaText.SetPosition(420, Height - MinimapHeight - 19);
+            ExpText.SetPosition(35, Height - MinimapHeight - 13);
+            GoldText.SetPosition(590, Height - MinimapHeight - 13);
+            PlatText.SetPosition(700, Height - MinimapHeight - 13);
+            HudSkillContainer.SetPosition(305, Height - 14);
+            HudSkillContainer.Relayout();
+            MapNameText.SetPosition(160, Height - 14);
         }
 
         public override void Update(float dt)
         {
             MessageWindow.Update(dt);
             FpsCounter.Update(dt);
+            MessageBoxContainer.Update(dt);
+            BacklogContainer.Update(dt);
+            DateText.Update(dt);
+            ClockHand.Update(dt);
         }
 
         public override void Draw()
         {
+            GraphicsEx.SetColor(Color.White);
+            MiniMapAsset.Draw(0, Height - MinimapHeight);
+            Minimap.Draw();
+            if (ShowingBacklog)
+            { 
+                BacklogBacking.Draw();
+                BacklogContainer.Draw();
+            }
+            GraphicsEx.SetColor(Color.White);
+            MessageBoxBacking.Draw();
+            HudBar.Draw();
             MessageWindow.Draw();
+            MessageBoxContainer.Draw();
+
+            GraphicsEx.SetColor(Color.White);
+            var barHeight = Height - MinimapHeight - 12;
+            ExpIconAsset.Draw(5, barHeight - 4);
+            ExpText.Draw();
+            if (!ShowingBacklog)
+            {
+                BarAsset.Draw(260, barHeight);
+                BarAsset.Draw(400, barHeight);
+                HealthText.Draw();
+                ManaText.Draw();
+                Love.Graphics.SetColor(Color.White);
+                var hpBarWidth = 83;
+                if (HpBarEntry != null)
+                    UiHelpers.DrawPercentageBar(HpBarEntry, new(277, barHeight + 5), HpBarEntry.HPRatio * hpBarWidth, Vector2i.Zero);
+                if (MpBarEntry != null)
+                    UiHelpers.DrawPercentageBar(MpBarEntry, new(417, barHeight + 5), MpBarEntry.HPRatio * hpBarWidth, new(hpBarWidth, 7));
+
+                GoldIconAsset.Draw(560, barHeight - 4);
+                PlatIconAsset.Draw(670, barHeight - 4);
+                GoldText.Draw();
+                PlatText.Draw();
+            }
+
             FpsCounter.Draw();
+            DateFrame.Draw(80, 8);
+            ClockAsset.Draw(0, 0);
```

Comment:
Can the clock elements be put in their own `UiElement`?

Reply:
changed in 9eb449f